src/core/datastore/persisters/key-value-persister.js

```diff
@@ -0,0 +1,67 @@
+const _cache = {};
+
+export class KeyValuePersister {
+  /** @type {Boolean} */
+  _cacheEnabled;
+  /** @type {Number} */
+  _ttl;
+
+  // TODO: implement TTL, make _cache a constructor argument?
+  constructor(cacheEnabled = true, ttl = Infinity) {
+    this._cacheEnabled = cacheEnabled;
+    this._ttl = ttl;
+  }
+
+  readEntities(key) {
+    if (this._cacheEnabled && _cache[key]) {
+      return Promise.resolve(_cache[key]);
+    }
+
+    return this._readFromPersistance(key)
+      .then((entities) => {
+        entities = entities || [];
```

Comment:
Instead of this you can do `((entities = []) => {`. This is stylistic but more concise.

Reply:
True. It wouldn't overwrite `null`, though. But I had plans to make persisters more "general-purpose" so they shouldn't overwrite `null` anyways. I'll make this change 👍

Reply:
I removed the default value as it's not the persister's job to decide what the default returned value should be. Whoever is using it should know better. There already is a default value in the inmemory repository (which is the only one using a KeyValuePersister for now) so no other change is necessary, I believe.